### PR TITLE
Luganda translations Mobile App

### PR DESCRIPTION
**Pull request by BenjaminSsempala**
**_WHAT DOES THIS PR DO?_**

**_WHAT ISSUES ARE RELATED TO THIS PR?_**

- Jira cards
    - []()
- GitHub issues
    - Closes #<enter GitHub issue number here>

**_HOW DO I TEST OUT THIS PR?_**

**_WHICH ENDPOINTS SHOULD BE READY FOR TESTING?:_** 

**_IS THERE ANY JENKINS CONSOLE LINK FOR CODE COVERAGE AND BUILD INFO?_**

**_ARE THERE ANY RELATED PRs?_**

### CODE REVIEW

**Review by Baalmart — COMMENTED** [review]
Thanks @BenjaminSsempala for this, much appreciated!! :) 

- Please review a recently merged PR #2908  that is fixing runtime errors associated with the translate UTIL.
- I gather that it might inspire some hotfixes in this very PR.

**Review by Baalmart — COMMENTED** [review]
Thanks @BenjaminSsempala , great additions made!

 I am also just leaving another edge case that might need further review.
**See attached screenshot.**
Screenshot 2024-01-17 at 18 04 16

**Comment by BenjaminSsempala** [comment]
> Thanks @BenjaminSsempala , great additions made!
> 
> I am also just leaving another edge case that might need further review. 

Yes, although wouldn't catching the error be desirable, Since these translations are inserted manually, in the case that the title or description, which are crucial, are missing, the errors would help catch that.

**Comment by Baalmart** [comment]
Thanks @BenjaminSsempala just left my response via Slack DM. 🤝

File: src/device-registry/utils/translate.js [hunk 1]
```diff
@@ -18,6 +18,15 @@ const translateUtil = {
       for (const healthTip of healthTips) {
         const translatedTip = { ...healthTip };
         const translations = await tipsTranslations(index, targetLanguage);
+
+        if (!translations.title || !translations.description) {
```
Comment (Baalmart) [inline src/device-registry/utils/translate.js:22]: Thanks @BenjaminSsempala , just left a followup Slack DM accordingly. Sincere apologies for dragging review thread.

File: src/device-registry/utils/translate.js [hunk 2]
```diff
@@ -96,35 +101,38 @@ const translateUtil = {
...
+            translatedQuestion.context = targetQuestion.context;
 
-            translatedAnswers.push(translatedAnswer);
-            answerIndex++;
+            const translatedAnswers = [];
+            let answerIndex = 0;
+            for (const answer of question.answers) {
+              const translatedAnswer = { ...answer };
+              const targetAnswer = targetQuestion.answers[answerIndex];
+              translatedAnswer.title = targetAnswer.title;
```
Comment (Baalmart) [inline src/device-registry/utils/translate.js:123]: Hi @BenjaminSsempala , thanks a lot for the recent changes. We could also do the same refactoring for the translated answers here.

**Review by Baalmart — COMMENTED** [review]
Hi @BenjaminSsempala , thanks again for this.

- I have just left a small comment for the `Answers` section of the same util.
- As a last feedback item that could be considered for future PRs or this one -- whatever works for you, we could also think about ways of optimising the entire translation process under each category by using `batching` as already demonstrated in the job located under `bin/store-readings-job` , starting on `code line 41 `of this script/job.

**Pull request merged**